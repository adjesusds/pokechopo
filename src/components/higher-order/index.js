export withTheme from './withTheme/withTheme'
