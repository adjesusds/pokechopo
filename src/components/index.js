export Button from './Button/Button'
export Copyright from './Copyright/Copyright'
export Divider from './Divider/Divider'
export Link from './Link/Link'
export Loading from './Loading/Loading'
export Message from './Message/Message'
export Root from './Root/Root'
