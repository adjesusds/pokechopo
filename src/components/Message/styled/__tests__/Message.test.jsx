// External Packages
import React from 'react'
import { create } from 'react-test-renderer'

// Internal Packages
import { withTheme } from 'components/higher-order'
import StyledMessage from '../Message'

describe('<StyledMessage />', () => {
  it('should style the component on valid props', () => {
    const StyledMessageWithTheme = withTheme(StyledMessage)

    let styledComponent

    styledComponent = create(<StyledMessageWithTheme />)

    expect(styledComponent).toMatchSnapshot()

    styledComponent = create(<StyledMessageWithTheme variant="danger" />)

    expect(styledComponent).toMatchSnapshot()

    styledComponent = create(<StyledMessageWithTheme variant="info" />)

    expect(styledComponent).toMatchSnapshot()

    styledComponent = create(<StyledMessageWithTheme variant="success" />)

    expect(styledComponent).toMatchSnapshot()

    styledComponent = create(<StyledMessageWithTheme variant="warning" />)

    expect(styledComponent).toMatchSnapshot()
  })
})
