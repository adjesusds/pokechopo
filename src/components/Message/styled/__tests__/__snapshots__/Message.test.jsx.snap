// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledMessage /> should style the component on valid props 1`] = `
.c0 {
  font-size: 1.25rem;
}

<span
  className="c0"
/>
`;

exports[`<StyledMessage /> should style the component on valid props 2`] = `
.c0 {
  color: #f44336;
  font-size: 1.25rem;
}

<span
  className="c0"
/>
`;

exports[`<StyledMessage /> should style the component on valid props 3`] = `
.c0 {
  color: #9e9e9e;
  font-size: 1.25rem;
}

<span
  className="c0"
/>
`;

exports[`<StyledMessage /> should style the component on valid props 4`] = `
.c0 {
  color: #4caf50;
  font-size: 1.25rem;
}

<span
  className="c0"
/>
`;

exports[`<StyledMessage /> should style the component on valid props 5`] = `
.c0 {
  color: #ffeb3b;
  font-size: 1.25rem;
}

<span
  className="c0"
/>
`;
